Frends.SFTP.ReadFile/Frends.SFTP.ReadFile/Frends.SFTP.ReadFile.cs

```diff
@@ -1,116 +1,182 @@
+        {
+            if (string.IsNullOrEmpty(connect.PrivateKeyString))
+                throw new ArgumentException("Private key string was not given.");
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(connect.PrivateKeyString));
+            privateKey = (connect.PrivateKeyFilePassphrase != null)
+                ? new PrivateKeyFile(stream, connect.PrivateKeyFilePassphrase)
+                : new PrivateKeyFile(stream, "");
+        }
+        switch (connect.Authentication)
+        {
+            case AuthenticationType.UsernamePassword:
+                methods.Add(new PasswordAuthenticationMethod(connect.UserName, connect.Password));
+                break;
+            case AuthenticationType.UsernamePrivateKeyFile:
+                methods.Add(new PrivateKeyAuthenticationMethod(connect.UserName, privateKey));
+                break;
+            case AuthenticationType.UsernamePasswordPrivateKeyFile:
+                methods.Add(new PasswordAuthenticationMethod(connect.UserName, connect.Password));
+                methods.Add(new PrivateKeyAuthenticationMethod(connect.UserName, privateKey));
+                break;
+            case AuthenticationType.UsernamePrivateKeyString:
+                methods.Add(new PrivateKeyAuthenticationMethod(connect.UserName, privateKey));
+                break;
+            case AuthenticationType.UsernamePasswordPrivateKeyString:
+                methods.Add(new PasswordAuthenticationMethod(connect.UserName, connect.Password));
+                methods.Add(new PrivateKeyAuthenticationMethod(connect.UserName, privateKey));
+                break;
+            default:
+                throw new ArgumentException($"Unknown Authentication type: '{connect.Authentication}'.");
+        }
+
+        connectionInfo = new ConnectionInfo(connect.Address, connect.Port, connect.UserName, methods.ToArray());
 
-            return Path.Combine(destination.Directory, file);
+        return connectionInfo;
+    }
+
+    /// <summary>
+    /// Get encoding for the file name to be transferred.
+    /// </summary>
+    private static Encoding GetEncoding(FileEncoding optionsFileEncoding, bool optionsEnableBom, string optionsEncodingInString)
+    {
+        switch (optionsFileEncoding)
+        {
+            case FileEncoding.UTF8:
+                return optionsEnableBom ? new UTF8Encoding(true) : new UTF8Encoding(false);
+            case FileEncoding.ASCII:
+                return Encoding.ASCII;
+            case FileEncoding.ANSI:
+                return Encoding.Default;
+            case FileEncoding.Unicode:
+                return Encoding.Unicode;
+            case FileEncoding.WINDOWS1252:
+                return Encoding.Default;
+            case FileEncoding.Other:
+                return Encoding.GetEncoding(optionsEncodingInString);
+            default:
+                throw new ArgumentOutOfRangeException();
```

Comment:
You should add some kind of custom error message to indicate what went wrong instead of generic error.